vector/v.out.ogr/export_areas_fast.c

```diff
@@ -0,0 +1,539 @@
+    }
+
+    if (donocat && doatt)
+        db_close_cursor(&cursor);
+
+    Vect_destroy_line_struct(Points);
+
+    return n_exported;
+}
+
+int export_areas_multi(struct Map_info *In, int field, int donocat,
+                       OGRFeatureDefnH Ogr_featuredefn, OGRLayerH Ogr_layer,
+                       struct field_info *Fi, dbDriver *driver, int ncol,
+                       int *colctype, const char **colname, int doatt,
+                       int nocat, int *n_noatt, int *n_nocat,
+                       int outer_ring_ccw)
+{
+    int i, n_exported, area, centroid;
+    int cat, last_cat, db_cat, line, findex, ipart;
+
+    struct line_pnts *Points;
+    struct line_cats *Cats;
+    struct ilist *line_list, *lcats;
+
+    struct Cat_index *ci;
+    int cat_index, n_cats;
+
+    dbString dbstring;
+    char buf[SQL_BUFFER_SIZE];
+    dbCursor cursor;
+    int more;
+    int key_col_index;
+
+    OGRGeometryH Ogr_geometry, Ogr_geometry_part;
+    OGRFeatureH Ogr_feature;
+    OGRwkbGeometryType wkbtype, wkbtype_part;
+
+    Points = Vect_new_line_struct();
+    Cats = Vect_new_cats_struct();
+    line_list = Vect_new_list();
+    lcats = Vect_new_list();
+
+    n_exported = 0;
+
+    /* check if category index is available for given field */
+    findex = Vect_cidx_get_field_index(In, field);
+    if (findex == -1) {
+        G_fatal_error(_("Unable to export multi-features. No category index "
+                        "for layer %d."),
+                      field);
+    }
+
+    ci = &(In->plus.cidx[findex]);
+    n_cats = ci->n_cats;
+
+    /* determine type */
+    wkbtype_part = wkbPolygon;
+    wkbtype = get_multi_wkbtype(wkbtype_part);
+
+    if (donocat)
+        G_message(_("Exporting features with category..."));
+
+    key_col_index = -1;
+    more = 1;
+    if (doatt) {
+        /* select attributes ordered by category value */
+        db_init_string(&dbstring);
+        sprintf(buf, "SELECT * FROM %s ORDER BY %s ASC", Fi->table, Fi->key);
+        G_debug(2, "SQL: %s", buf);
+        db_set_string(&dbstring, buf);
+        if (db_open_select_cursor(driver, &dbstring, &cursor, DB_SEQUENTIAL) !=
+            DB_OK) {
+            G_fatal_error(_("Cannot select attributes sorted by %s"), Fi->key);
+        }
+
+        if (db_fetch(&cursor, DB_NEXT, &more) != DB_OK)
+            G_fatal_error(_("Unable to fetch data from table"));
+
+        /* get index of key column */
+        key_col_index = -1;
+        for (i = 0; i < ncol; i++) {
+            if (strcmp(Fi->key, colname[i]) == 0) {
+                key_col_index = i;
+                break;
+            }
+        }
+    }
+
+    last_cat = -1;
+    db_cat = -1;
+    cat_index = 0;
+    while (cat_index < n_cats) {
+
+        G_percent(cat_index, n_cats, 5);
+
+        cat = ci->cat[cat_index][0];
+
+        /* make sure the cidx is ordered by cat */
+        if (cat < last_cat)
+            G_fatal_error("Categor index is not sorted ascending by cat!");
```

Comment:
```suggestion
            G_fatal_error("Category index is not sorted ascending by cat!");
```
`Categor` isn't supposed to be there?